driver-core/src/main/java/com/datastax/driver/core/ArrayBackedRow.java

```diff
@@ -239,6 +242,23 @@ public InetAddress getInet(String name) {
         return getInet(metadata.getFirstIdx(name));
     }
 
+    public Token getToken(int i) {
+        if (tokenFactory == null)
+            throw new DriverInternalError("Token factory not set. This should only happen at initialization time");
+
+        metadata.checkType(i, tokenFactory.getTokenType().getName());
+
+        ByteBuffer value = data.get(i);
+        if (value == null || value.remaining() == 0)
+            return null;
+
+        return tokenFactory.deserialize(value);
+    }
+
+    public Token getToken(String name) {
+        return getToken(metadata.getFirstIdx(name));
```

Comment:
> I did eventually implement getToken("i") per your suggestion

Yes, but what I'm saying is that my suggestion is actually broken for compound partition keys, which is really not ideal.

> my impression is that if you use ? placeholders you tend to use index-based setters, so the weird column name wouldn't be that big of a problem

I'm not sure where that impression comes from but I strongly suspect its wrong, if only because there is plenty of example on the web that use `?` placeholders and name-based setters. And for normal columns that combination works pretty well. In fact, I rarely see people using index-based setters (because it's not terribly friendly), and I don't see everyone using named bind markers. Anyway, I don't have numbers to back any of this but I have a strong hunch that you are overly optimistic on that front.

Anyway, I do worry that having non-symetric `getToken` and `setToken` will seriously surprise people and break expectations.